packages/block-library/src/social-links/style.scss

```diff
@@ -95,14 +95,23 @@
 }
 
 // This needs specificity because themes usually override it with things like .widget-area a.
-.wp-block-social-links .wp-block-social-link .wp-block-social-link-anchor {
-	&,
-	&:hover,
-	&:active,
-	&:visited,
-	svg {
-		color: currentColor;
-		fill: currentColor;
+.wp-block-social-links .wp-block-social-link.wp-social-link {
+	margin: 0;
+	padding: 0;
+
+	.wp-block-social-link-anchor {
+		&,
+		&:hover,
+		&:active,
+		&:visited,
+		svg {
+			color: currentColor;
+			fill: currentColor;
+		}
+	}
+
+	&:after {
```

Comment:
Hey @jasmussen - no, the `padding` and `margin` is the main part of the fix which covers most themes. However, at least two themes which I tested also set an `:after` in the widget area (eg. see Apostrophe 2 in Automattic/themes#7062).

I feel that more than enough themes are affected to justify the `padding` and `margin` styles in Core, but I don’t mind particularly either way on this bit. However, since we already have a selector, I suspect that it’s probably easier for users and developers than hoping that themes update their styling for this block. 

I hope that makes sense, and thanks for the review! :)

Reply:
The fix is definitely valid, my question is mostly around this pseudo element. Do you know where this code is coming from?

![Screenshot 2023-12-01 at 10 51 08](https://github.com/WordPress/gutenberg/assets/1204802/ede3907e-cffd-42e5-b49b-5b2f3e469967)

If it's reproducible in core on more than just one theme, for example with social icons in widgets on every theme that has widget areas, then the fix is definitely appropriate for core.

But if the CSS that appends a | is coming from a theme or a plugin, then it should not be fixed here, it should be fixed at the source. Makes sense?

Reply:
That makes sense! A few themes are guilty of similar styling, but yes, it's from the theme: https://github.com/Automattic/themes/blob/012258cc65cfdfc99bc32117ef951f2c9dbb7189/apostrophe-2/style.css#L1411-L1415

I don't have a strong view either way as to whether there's enough themes affected to justify a Core fix, so I've removed that style from this PR - I'll fix it in the individual themes instead. :)

Reply:
Thanks for removing. It's definitely important to not fix at the source, because like I said in some cases we use the pseudo selectors for other things, so this'll have to be a local fix.

Reply:
Makes total sense! Please let me know if there are any issues with the remaining changes. :)

Reply:
Looks good, hopefully @apeatling can give it a green check!